include/targets/cuda/reduce_helper.h

```diff
@@ -59,6 +59,7 @@ namespace quda
      memory
    */
   template <typename T, bool use_kernel_arg = true> struct ReduceArg : kernel_param<use_kernel_arg> {
+    using reduce_t = T;
```

Comment:
The changes in this file will need to be propagated unto `include/targets/generic/reduce_helper.h`

Reply:
2604207eb931edd2da6efb1768323aa2158e905b

Reply:
and this: 960573f1a94b93e87bc4282c0e56aef986eb18ce